fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/eventlist/EventListAdapter.java

```diff
@@ -7,79 +7,62 @@
 
 import java.util.*;
-import java.util.stream.Collectors;
 
 import io.reactivex.Observable;
 
-import static com.github.dedis.popstellar.model.objects.event.EventCategory.*;
-import static com.github.dedis.popstellar.ui.detail.LaoDetailActivity.setCurrentFragment;
+import static com.github.dedis.popstellar.model.objects.event.EventCategory.PAST;
+import static com.github.dedis.popstellar.model.objects.event.EventCategory.PRESENT;
 
-public class EventListAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+public class EventListAdapter extends EventsAdapter {
 
-  private final LaoDetailViewModel viewModel;
-  private final FragmentActivity activity;
   private final EnumMap<EventCategory, List<Event>> eventsMap;
 
-  private final boolean[] expanded = new boolean[3];
+  private final boolean[] expanded = new boolean[2];
   public static final int TYPE_HEADER = 0;
   public static final int TYPE_EVENT = 1;
   public static final String TAG = EventListAdapter.class.getSimpleName();
 
   public EventListAdapter(
       LaoDetailViewModel viewModel, Observable<Set<Event>> events, FragmentActivity activity) {
+    super(events, viewModel, activity, TAG);
     this.eventsMap = new EnumMap<>(EventCategory.class);
     this.eventsMap.put(PAST, new ArrayList<>());
     this.eventsMap.put(PRESENT, new ArrayList<>());
-    this.eventsMap.put(FUTURE, new ArrayList<>());
-    this.activity = activity;
-    this.viewModel = viewModel;
 
     expanded[PAST.ordinal()] = true;
     expanded[PRESENT.ordinal()] = true;
-    expanded[FUTURE.ordinal()] = true;
-
-    subscribeToEventSet(events);
-  }
-
-  private void subscribeToEventSet(Observable<Set<Event>> observable) {
-    this.viewModel.addDisposable(
-        observable
-            .map(events -> events.stream().sorted().collect(Collectors.toList()))
-            // No need to check for error as the events errors already handles them
-            .subscribe(this::putEventsInMap, err -> Log.d(TAG, "ERROR", err)));
   }
 
-  /** A helper method that places the events in the correct key-value pair according to state */
+  /**
+   * A helper method that places the events in the correct key-value pair according to state Closed
+   * events are put in the past section. Opened events in the current section. Created events that
+   * are to be open in less than 24 hours are also in the current section. Created events that are
+   * to be opened in more than 24 hours are not displayed here (They are displayed in a separate
+   * view)
+   */
   @SuppressLint("NotifyDataSetChanged")
-  private void putEventsInMap(List<Event> events) {
+  @Override
+  public void updateEventSet(List<Event> events) {
     this.eventsMap.get(PAST).clear();
-    this.eventsMap.get(FUTURE).clear();
     this.eventsMap.get(PRESENT).clear();
 
     for (Event event : events) {
       switch (event.getState()) {
         case CREATED:
-          eventsMap.get(FUTURE).add(event);
+          if (event.isEventEndingToday()) {
```

Comment:
today as in less than a day ?

Reply:
Yes. I changed the doc of the function. Do you have a better idea for naming ?